src/graphql/root/subscription/price.ts

```diff
@@ -39,34 +40,36 @@ const PriceSubscription = {
     const { amount, amountCurrencyUnit, priceCurrencyUnit } = args.input
 
     const eventName = SAT_USDCENT_PRICE
+    const immediateEventName = `SAT_USDCENT_PRICE_${uuidv4()}`
 
     for (const input of [amountCurrencyUnit, priceCurrencyUnit]) {
       if (input instanceof Error) {
-        pubsub.publishImmediate(eventName, {
+        pubsub.publishImmediate(immediateEventName, {
           errors: [{ message: input.message }],
         })
-        return pubsub.asyncIterator(eventName)
+        return pubsub.asyncIterator(immediateEventName)
       }
     }
 
     if (amountCurrencyUnit !== "BTCSAT" || priceCurrencyUnit !== "USDCENT") {
       // For now, keep the only supported exchange price as SAT -> USD
-      pubsub.publishImmediate(eventName, {
+      pubsub.publishImmediate(immediateEventName, {
         errors: [{ message: "Unsupported exchange unit" }],
       })
     } else if (amount >= 1000000) {
       // SafeInt limit, reject for now
-      pubsub.publishImmediate(eventName, {
+      pubsub.publishImmediate(immediateEventName, {
         errors: [{ message: "Unsupported exchange amount" }],
       })
     } else {
       const satUsdPrice = await Prices.getCurrentPrice()
       if (!(satUsdPrice instanceof Error)) {
-        pubsub.publishImmediate(eventName, { satUsdCentPrice: 100 * satUsdPrice })
+        pubsub.publishImmediate(immediateEventName, { satUsdCentPrice: 100 * satUsdPrice })
       }
+      return pubsub.asyncIterator([immediateEventName, eventName])
     }
 
-    return pubsub.asyncIterator(eventName)
+    return pubsub.asyncIterator(immediateEventName)
```

Comment:
I think this one should have both events.

Reply:
I think it needs to just be the immediateEventName because in the case that the user received an immediate error for wrong inputs, they should continue to receive data for something they didnt ask for

Reply:
I see but that means the subscription will only work if there's a valid initial response from the Prices repo, and would otherwise fail. Might be okay.

Reply:
Its not inside the if statement, so I think even if the first price repo call is invalid it will still work in the future.